### DISCUSSION TITLE
BOM Validation: RFC 3987 External References

### DISCUSSION THREAD

**Discussion by setchy** [DependencyTrack/dependency-track#3775] [Q&A] [answered]
DTrack: 4.11.1

We've seen a handful of BOMs (generated by cdxgen) being rejected by the new BOM Validation Process due to invalid URLs (or IRI References)

Example below
```
"externalReferences": [
      {
          "type": "vcs",
          "url": "https://github.com/follow-redirects/follow-redirects"
      },
      {
          "type": "vcs",
          "url": "git@github.com:follow-redirects/follow-redirects.git"
      }
  ],
 ```

The second reference is the one that doesn't pass.

I'd like to confirm that this is an expected rejection before I look to do any changes within cdxgen

**Comment by nscuro** [comment, marked as answer]
That's not something DT decides, this is how it's defined in the CycloneDX schema: https://cyclonedx.org/docs/1.5/json/#components_items_externalReferences_items_url

CC @stevespringett @jkowalleck

**Reply by nscuro** [reply]
The `iri-reference` format of JSON Schema is described here: https://json-schema.org/understanding-json-schema/reference/string#resource-identifiers

**Reply by setchy** [reply]
Thanks @nscuro - that's all i needed :) 

I'll look to fix upstream

**Reply by jkowalleck** [reply]
> I'll look to fix upstream 

I'd suggest approaching the cdxgen project via bug report, and ask them to use <https://www.npmjs.com/package/hosted-git-info> to get the git urls fixed. 

This worked for other JavaScript-based CycloneDX generators.

**Reply by setchy** [reply]
Thanks @jkowalleck
